### DISCUSSION TITLE
Can clickhouse take benefit of partitioned data in s3?

### DISCUSSION THREAD

**Discussion by gauravphoenix** [ClickHouse/ClickHouse#27711] [General]
Let's say I have data in s3 like this -
```
s3://bucket/customer1/year/month/day/foo.json
s3://bucket/customer2/year/month/day/bar.json

```

and if I were to run query `select * from foo where customer_id= customer1`, can clickhouse only read the `s3://bucket/customer1/year/month/day/foo.json` path and skip `s3://bucket/customer2` ? 

The s3 table engine [documentation](https://clickhouse.tech/docs/en/engines/table-engines/integrations/s3/) doesn't have partition example so I can't figure out if it is possible or not.

**Comment by alexey-milovidov** [comment]
Yes, there is `_path` and `_file` virtual columns that you can use for filtering.
